development/extensions/tutorial_notifications.rst

```diff
@@ -0,0 +1,955 @@
+Mark notifications as read when necessary, even though in a lot of cases this is done automatically.
+The user has a centralised place for all notifications, which is easily accessible.
+
+By default there are two methods of notification.
+Namely per email or per board notification.
+The email method is only available when the ''board-wide emails'' setting has been enabled.
+Extensions can also create and add their own notification methods.
+
+File structure
+--------------
+This is an overview of the file structure needed for a notification to function within an extension.
+In total there are 7 files that will be covered in this tutorial, to create and send notifications.
+These files will be mentioned throughout this tutorial.
+
+
+| |fa-user| vendor
+| └─ |fa-ext| extension
+|  |nbsp| ├─ |fa-folder| config
+|  |nbsp| │ |nbsp| └─ |fa-file| services.yml
+|  |nbsp| ├─ |fa-folder| event
+|  |nbsp| │ |nbsp| └─ |fa-file| listener.php
+|  |nbsp| ├─ |fa-folder| language
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| en
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-folder| email
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| short
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| extension_common.php
+|  |nbsp| ├─ |fa-folder| notification
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| type
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| sample.php
+|  |nbsp| └─ |fa-file| ext.php
+|
+
+Creating a notification type
+============================
+The notification type that will be used in this tutorial is: ``vendor.extension.notification.type.sample``.
+
+Registering as a Symfony service
+--------------------------------
+Firstly the notification type has to be registered as a Symfony service.
+As with any service declaration, it will need a service identifier and the class argument.
+However, notifications types also need three additional parameters. The parent, shared and tags parameters.
+The parent parameter will indicate from which service this notification type will inherit.
+And the shared parameter will make sure that each time this notification type is requested, a new instance is created.
+This will prevent that any data will be transferred over.
+Lastly, the service needs to be tagged as a notification type, so it will be correctly listed in all places.
+
+.. code-block:: yaml
+   :caption: :class:`vendor/config/services.yml`
+
+   services:
+       vendor.extension.notification.type.sample:
+           class: vendor\extension\notification\type\sample
+           parent: notification.type.base
+           shared: false
+           tags: [{ name: notification.type }]
+
+However, we want to use two additional services in our notification type.
```

Comment:
Maybe explain briefly why we need this other classes and what they will be used for.